test/e2e/functional_test.go

```diff
@@ -84,6 +84,31 @@ spec:
 		ExpectWorkflowDeleted()
 }
 
+func (s *FunctionalSuite) TestWorkflowRetention() {
+	var listOptions metav1.ListOptions
+	for i := 0; i < 2; i++ {
+		s.Given().
+			Workflow(`
+metadata:
+  generateName: workflow-retention-
+spec:
+  ttlStrategy:
+    secondsAfterCompletion: 86400
+  entrypoint: main
+  templates:
+    - name: main
+      container:
+        image: argoproj/argosay:v2
+`).When().SubmitWorkflow().SubmitWorkflow().WaitForWorkflow(fixtures.ToBeSucceeded, time.Minute)
+	}
+	s.Given().When().
+		Wait(3*time.Second).
```

Comment:
can we please avoid Wait?

Reply:
> we're nearly there, I think the fundamentals are now solid, noodling on implementation details
> 
> great stuff - I expect we'll use this

Thanks for the feedback, will continue to work on this tomorrow!